### PR TITLE
fix: (platform) Create common base class for Platform components

### PR DESCRIPTION
**Pull request by DeepakSap14**
#### Please provide a link to the associated issue.
https://github.com/SAP/fundamental-ngx/issues/2368

#### Please provide a brief summary of this pull request.
In Platform we have multiple components with some common properties eg. id, name, placeholder etc. these can be moved to a common class and can be extended by all components being developed or to be developed. This PR includes base class and changes to existing components to extend this base class.

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [x] tests for the changes that have been done

Documentation checklist:
- [x] update `README.md`
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

File: libs/platform/src/lib/components/base.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    protected _disabled: boolean;
+
+    @Input()
+    id: string = this.defaultId;
+
+    @Input()
+    name: string;
+
+    @Input()
+    placeholder: string;
```
Comment (fkolar) [inline libs/platform/src/lib/components/base.ts:22]: Placeholder makes sense  only for inputs, unless you can tell me o the situatoins

File: libs/platform/src/lib/components/base.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    @Input()
+    contentDensity: InputSize = 'cozy';
```
Comment (fkolar) [inline libs/platform/src/lib/components/base.ts:25]: The reason why we renamed bindings is to also fix naming for InputSize that does not make sense, Rename inputsize to ContentDensity

File: libs/platform/src/lib/components/base.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+ * this can be extended to reduce the code duplication across components.
+ * @hidden for form related Base , see BaseInput.
+ */
+export class BaseComponent {
+    protected defaultId: string = `fdp-id-${randomId++}`;
+    protected _disabled: boolean;
+
+    @Input()
+    id: string = this.defaultId;
+
```
Comment (fkolar) [inline libs/platform/src/lib/components/base.ts:17]: also add in there Inputs(), called width

File: libs/platform/src/lib/components/base.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+import { InputSize } from './form/form-control';
+
+let randomId = 0;
+
+/**
+ * This class contains common properties used across components. 
+ * this can be extended to reduce the code duplication across components.
+ * @hidden for form related Base , see BaseInput.
+ */
+export class BaseComponent {
```
Comment (fkolar) [inline libs/platform/src/lib/components/base.ts:11]: make this class abstract

File: libs/platform/src/lib/components/form/form-control.ts [hunk 5]
```diff
@@ -30,7 +30,7 @@ export abstract class FormFieldControl<T> {
     /**
      *  Components works in two sizes compact or cozy
      */
-    size: InputSize;
+    contentDensity: InputSize;
```
Comment (fkolar) [inline libs/platform/src/lib/components/form/form-control.ts:33]: see my other comment

Comment (DeepakSap14) [reply libs/platform/src/lib/components/form/form-control.ts]: have pushed changes with reviews implemented. please check.

**Comment by fkolar** [comment]
> I'm not sure `BaseComponent` should contain the `width` and `disabled` properties. These properties pertain to inline elements and input elements. I don't think they'll be used for components like ActionBar or List.

We should see if there are other cases for this: 
- Alert
- button
- etc..

If we cannot see other usecase, even I think container could have width and could be disabled, then let;s remove it, we can add it back anything

**Comment by DeepakSap14** [comment]
Merging it after master rebase.

**Pull request merged**